source/MaterialXContrib/MaterialXNode/MaterialXNode.h

```diff
@@ -26,6 +26,7 @@ class MaterialXNode : public MPxNode
     bool setInternalValue(const MPlug&, const MDataHandle&) override;
 
     void setData(const MString& documentFilePath, const MString& elementPath, std::unique_ptr<MaterialXData>&&);
+    void createAndRegisterFragment(bool reloadDocument=false);
```

Comment:
Maybe this method could be made private and there could be a public reloadDocument() method that would reset both the document and the MaterialXData and then call this method? Then reloadDocument() could also be called when setting the internal value of the document path attribute.